backend/src/server/app.ts

```diff
@@ -87,7 +87,16 @@ export const main = async ({ db, hsmModule, auditLogDb, smtp, logger, queue, key
 
     await server.register<FastifyCorsOptions>(cors, {
       credentials: true,
-      origin: appCfg.SITE_URL || true
+      ...(appCfg.CORS_ALLOWED_ORIGINS?.length
+        ? {
+            origin: appCfg.CORS_ALLOWED_ORIGINS
+          }
+        : {
+            origin: appCfg.SITE_URL || true
```

Comment:
the site url should always be apart of the list og origins even if CORS_ALLOWED_ORIGINS is specified? Otherwise the frontend wouldn't work. We want that to always work